libraries/AP_ExternalAHRS/AP_ExternalAHRS.cpp

```diff
@@ -240,6 +241,29 @@ bool AP_ExternalAHRS::pre_arm_check(char *failure_msg, uint8_t failure_msg_len)
         hal.util->snprintf(failure_msg, failure_msg_len, "ExternalAHRS: No origin");
         return false;
     }
+
+    // Verify the user has configured the GPS to accept EAHRS data.
+    if (has_sensor(AvailableSensor::GPS)) {
+        const auto eahrs_gps_sensors = backend->num_gps_sensors();
+
+        if (AP::gps().num_sensors() < eahrs_gps_sensors) {
```

Comment:
So this might amuse you, but `num_sensors` here doesn't do what you think it does in the case that `GPS1_TYPE` is 0 and `GPS2_TYPE1` is 21.  It was like that when I got here.

I suggest you omit this check.

Reply:
Ok. the second one should cover it then. I'll make sure CI tests some cases.

Reply:
CI for asserting that a certain pre-arm fails is not built yet. If possible, can we leave CI out of scope for this?

Reply:
Done!